apps/web/src/components/tx/ConfirmTxDetails/NameChip.tsx

```diff
@@ -13,24 +14,27 @@ const NameChip = ({
   txInfo?: TransactionDetails['txInfo']
   withBackground?: boolean
 }) => {
+  const addressBook = useAddressBook()
   const toAddress = txData?.to.value
   const customTxInfo = txInfo && isCustomTxInfo(txInfo) ? txInfo : undefined
   const toInfo = customTxInfo?.to || txData?.addressInfoIndex?.[txData?.to.value] || txData?.to
-  const toName = toInfo?.name || (toInfo && 'displayName' in toInfo ? String(toInfo.displayName || '') : undefined)
+  const nameFromAb = toAddress !== undefined ? addressBook[toAddress] : undefined
+  const toName =
+    toInfo?.name || (toInfo && 'displayName' in toInfo ? String(toInfo.displayName || '') : undefined) || nameFromAb
```

Comment:
`nameFromAb` should probably come first.

Reply:
Changed priorities and added a test case for this.